projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.ts

```diff
@@ -60,38 +55,23 @@ export class ResourceViewComponent implements OnInit, OnChanges {
       (response: ReadResource) => {
         this.resource = response;
 
-        // get list of all properties
-        const propsList: IHasProperty[] = this.resource.entityInfo.classes[this.resource.type].propertiesList;
-
-        for (const prop of propsList) {
-          const index = prop.propertyIndex;
-
-          if (this.resource.entityInfo.properties[index]) {
-            if (this.resource.entityInfo.properties[index] instanceof ResourcePropertyDefinition) {
-              // filter all properties by type ResourcePropertyDefinition
-              const propInfoAndValues: PropertyInfoValues = {
-                guiDef: prop,
-                propDef: this.resource.entityInfo.properties[index],
-                values: this.resource.properties[index]
-              };
-
-              this.propArray.push(propInfoAndValues);
-
-            } else if (this.resource.entityInfo.properties[index] instanceof SystemPropertyDefinition) {
-              // filter all properties by type SystemPropertyDefinition
-              const systemPropInfo = this.resource.entityInfo.properties[index];
-
-              this.systemPropArray.push(systemPropInfo);
-
+        // gather resource property information
+        this.resPropInfoVals = this.resource.entityInfo.classes[this.resource.type].getResourcePropertiesList().map(
+            (prop: IHasPropertyWithPropertyDefinition) => {
+                const propInfoAndValues: PropertyInfoValues = {
+                    propDef: prop.propertyDefinition,
+                    guiDef: prop,
+                    values: this.resource.getValues(prop.propertyIndex)
+                };
+                return propInfoAndValues;
             }
-
-          } else {
-            console.error('Error detected: the property with IRI =' + index + 'is not a property of the resource');
-          }
-        }
+        );
 
         // sort properties by guiOrder
-        this.propArray.sort((a, b) => (a.guiDef.guiOrder > b.guiDef.guiOrder) ? 1 : -1);
+        this.resPropInfoVals.sort((a, b) => (a.guiDef.guiOrder > b.guiDef.guiOrder) ? 1 : -1);
+
+        // get system property information
+        this.systemPropDefs = this.resource.entityInfo.getPropertyDefinitionsByType(SystemPropertyDefinition);
 
       },
       (error: ApiResponseError) => {
```

Comment:
Good that you check for unsuccessful responses, I think we should apply such a pattern throughout the lib / app.